apps/contribution/test/contribution.js

```diff
@@ -0,0 +1,349 @@
+    });
+  });
+
+  describe("Add contribution", async () => {
+    let amount = 100;
+    let contributorId = 1;
+    let hashDigest = '0x0000000000000000000000000000000000000000000000000000000000000000';
+    let hashFunction = 1;
+    let hashSize = 1;
+
+    it("should revert when add contribution from address that does not have permission", async () => {
+      return assertRevert(async () => {
+        await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: member1});
+        'sender does not have permission';
+      });
+    });
+
+    it("should revert when add contribution with amount equal to zero", async () => {
+      return assertRevert(async () => {
+        await contribution.add(0, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+        'amount equal to zero';
+      });
+    });
+
+    it("should add contribution", async () => {
+      let contributionCountBefore = await contribution.contributionsCount();
+      await contribution.add(amount, contributorId, hashDigest, hashFunction, hashSize, {from: root});
+      let contributionCountAfter = await contribution.contributionsCount();
+      // eslint-disable-next-line no-undef
+      assert.equal(contributionCountAfter.toNumber()-contributionCountBefore.toNumber(), 1, "contributions counter incremented");
+      let contributionObject = await contribution.getContribution(contributionCountAfter.toNumber());
+      // eslint-disable-next-line no-undef
+      assert.equal(contributionObject[1], contributorId, "contribution added belong to contributor id");
+      let isExist = await contribution.exists(contributionCountAfter.toNumber());
+      // eslint-disable-next-line no-undef
+      assert.equal(isExist, true, "contribution exist");
+    });
+  });
+
+  describe("Veto contribution", async () => {
+    it("should revert when veto from address that does not have permission", async () => {
+      const contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber(), {from: member1});
+        'sender does not have permission to veto';
+      });
+    });
+
+    it("should revert when veto contribution that does not exist", async () => {
+      const contributionId = await contribution.contributionsCount();
+      return assertRevert(async () => {
+        await contribution.veto(contributionId.toNumber()+1, {from: root});
+        'contribution not found';
+      });
+    });
+
+    it("veto contribution", async () => {
+      const contributionId = await contribution.contributionsCount();
+      let contributionObject = await contribution.getContribution(contributionId.toNumber());
+      console.log("veto block: " + contributionObject[7]);
+      console.log("current block: " + await getBlockNumber());
+
+      await contribution.veto(contributionId.toNumber(), {from: root});
+      // eslint-disable-next-line no-undef
```

Comment:
blocks get minted and the veto period end right away... @bumi  @skddc

Reply:
Yes, the first 10 are confirmed immediately iirc.